node/forest_libp2p/src/service.rs

```diff
@@ -398,6 +292,458 @@ where
+            peer_id,
+            request,
+            response_channel,
+        } => {
+            let request_id = st_mut
+                .behaviour_mut()
+                .chain_exchange
+                .send_request(&peer_id, request);
+            cx_request_table.insert(request_id, response_channel);
+            emit_event(
+                network_sender_out,
+                NetworkEvent::ChainExchangeRequestOutbound { request_id },
+            )
+            .await;
+        }
+        NetworkMessage::BitswapRequest {
+            cid,
+            response_channel: _,
+        } => match st_mut.behaviour_mut().want_block(cid) {
+            Ok(query_id) => {
+                outgoing_bitswap_query_ids.insert(query_id, cid);
+                emit_event(
+                    network_sender_out,
+                    NetworkEvent::BitswapRequestOutbound { query_id, cid },
+                )
+                .await;
+            }
+            Err(e) => warn!("Failed to send a bitswap want_block: {}", e.to_string()),
+        },
+        NetworkMessage::JSONRPCRequest { method } => match method {
+            NetRPCMethods::NetAddrsListen(response_channel) => {
+                let listeners: Vec<_> = Swarm::listeners(st_mut).cloned().collect();
+                let peer_id = Swarm::local_peer_id(st_mut);
+
+                if response_channel.send((*peer_id, listeners)).is_err() {
+                    warn!("Failed to get Libp2p listeners");
+                }
+            }
+            NetRPCMethods::NetPeers(response_channel) => {
+                let peer_addresses: &HashMap<PeerId, Vec<Multiaddr>> =
+                    st_mut.behaviour_mut().peer_addresses();
+
+                if response_channel.send(peer_addresses.to_owned()).is_err() {
+                    warn!("Failed to get Libp2p peers");
+                }
+            }
+            NetRPCMethods::NetConnect(response_channel, peer_id, mut addresses) => {
+                let mut success = false;
+
+                for multiaddr in addresses.iter_mut() {
+                    multiaddr.push(Protocol::P2p(
+                        Multihash::from_bytes(&peer_id.to_bytes()).unwrap(),
+                    ));
+
+                    if Swarm::dial(st_mut, multiaddr.clone()).is_ok() {
+                        success = true;
+                        break;
+                    };
+                }
+
+                if response_channel.send(success).is_err() {
+                    warn!("Failed to connect to a peer");
+                }
+            }
+            NetRPCMethods::NetDisconnect(response_channel, peer_id) => {
+                let _ = Swarm::disconnect_peer_id(st_mut, peer_id);
+                if response_channel.send(()).is_err() {
+                    warn!("Failed to disconnect from a peer");
+                }
+            }
+        },
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn handle_forest_behaviour_event<DB, P: StoreParams>(
```

Comment:
It would be nice to refactor such oversized methods so there's a single `match` calling methods. 300 LoC for a method isn't ideal.